Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -5449,8 +6047,10 @@ class BBQJIT {
             BLOCK(
                 if (m_jit.supportsCountPopulation())
                     m_jit.countPopulation32(operandLocation.asGPR(), resultLocation.asGPR());
-                else
-                    emitCCall(&operationPopcount32, Vector<Value> { operand }, TypeKind::I32, result);
+                else {
+                    result = topValue(TypeKind::I32);
```

Comment:
The `EMIT_UNARY(...)` macro will already assign `result` to the top value on the stack and give it a register, so it should be able to be passed in. However, this creates a somewhat nasty tacit dependency - `emitCCall` will bind `result` to the `returnValueGPR`, which will error if `result` is already bound to a different register (to avoid mucking with the register allocator state). It shouldn't currently error specifically because we only allocate caller-saved registers, which get flushed across the call regardless; however, if we add callee-saved register allocation to BBQJIT in the future, we could get very niche errors.

All this is to say, I think the most correct thing with the current abstractions would be to
```
consume(result);
emitCCall(&operationPopcount32, Vector<Value> { operand }, result);
```
so `result` is quietly unbound from its current register before we pass it into `emitCCall`. This only applies to `addI32Popcnt` and `addI64Popcnt` since they are currently the only functions that use `EMIT_UNARY` and contain a C call.

Reply:
I'll make the change to consume. Unfortunately I think we won't have any bots that test this code path anyway. Since I don't think we have any bots that run on old enough CPUs that they'll fail the `supportsCountPopulation()` test now that arm64 always passes. (For reference the first Intel processor with POPCNT came out 14 years ago https://en.wikipedia.org/wiki/Nehalem_(microarchitecture) and I think earlier for AMD)